### DISCUSSION TITLE
Edge Runtime and batched job scheduling

### DISCUSSION THREAD

**Discussion by fdecampredon** [vercel/next.js#44643] [Help]
We are trying to use edge-runtime for our GraphQL api route.
Inside of our resolvers, We are using [DataLoader](https://github.com/graphql/dataloader#batch-scheduling), for caching and **batching** purpose.

The fact that Edge Runtime doesn't provide either `process.nextTick` nor `setImmediate` force the usage of `setTimeout` to schedule batched jobs.
However the minimum 4ms of delay in `setTimeout` can ends up creating slower response, especially in case of large and deep GraphQL queries.
Is there a better alternative to schedule a job after all microtask are executed ?

**Comment by sgrund14** [comment]
hey @fdecampredon, did you guys end up figuring this out?
